ale/base/data_naif.py

```diff
@@ -0,0 +1,198 @@
+import spiceypy as spice
+import numpy as np
+from ale.base.type_sensor import Framer
+
+class NaifSpice():
+    def __enter__(self):
+        """
+        Called when the context is created. This is used
+        to get the kernels furnished.
+        """
+        if self.metakernel:
+            spice.furnsh(self.metakernel)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Called when the context goes out of scope. Once
+        this is done, the object is out of scope and the
+        kernels can be unloaded.
+        """
+        spice.unload(self.metakernel)
+
+    @property
+    def metakernel(self):
+        pass
+
+    @property
+    def _odtx(self):
+        """
+        Returns
+        -------
+        : list
+          Optical distortion x coefficients
+        """
+        return spice.gdpool('INS{}_OD_T_X'.format(self.ikid),0, 10).tolist()
+
+    @property
+    def _odty(self):
+        """
+        Returns
+        -------
+        : list
+          Optical distortion y coefficients
+        """
+        return spice.gdpool('INS{}_OD_T_Y'.format(self.ikid), 0, 10).tolist()
+
```

Comment:
We pretty much assume they do exist. If an instrument has a slightly different variation on this they would have to override with different key/array length. If you try to use this method against an instrument with a different distortion model it will naturally except with naif key not found signalling to the developer that the distortion model is probably different. Unless I am misunderstanding something?